recipes-core/rcu-hostname/files/rcuhostname

```diff
@@ -0,0 +1,30 @@
+#!/bin/sh
+
+# This script is responsible for reading the RCU serial and setting the RCU hostname accordingly. Paired with avahi-daemon which enables mDNS, this provides an option for clients to communicate with the RCU just by knowing its serial (instead of IP address).
+
+# TODO: Change RCU serial placeholder path
+rcuserialpath=/usr/bin/rcu-serial
+tdxserialpath=/proc/device-tree/serial-number
+
+#set serial to Apalis SOM serial number
+if [ -f ${tdxserialpath} ]; then
+    serial=$(tr -d '\0' <${tdxserialpath})
+fi
+
+#if RCU serial exists, we prefer the RCU serial number
+if [ -f ${rcuserialpath} ]; then
+    serial=$(tr -d '\0' <${rcuserialpath})
+fi
+
+#if serial number is still empty
+if [ -z "$serial" -a "$serial" != " "  ]; then
+    serial="EMPTYSERIAL"
+fi
+
+hostname="NI-SmartRack-"${serial}
```

Comment:
Refer to Jonathan's comment on the feature spec, we need a way for user to set their custom hostname. 
I believe user will set the hostname via RCU Service, perhaps you need some placeholder code to read from a file (where RCU Service will write the user-defined hostname). If this file exist and the content is valid, we will use it as the hostname instead.

Reply:
@karhin1990 I added a todo in this file, as well as a new task under the RCU image user story in AzDO.